tests/unit/test_topology.py

```diff
@@ -0,0 +1,212 @@
+# pylint: disable=no-self-use, redefined-outer-name
+
+from copy import deepcopy
+
+import pytest
+
+from expertsystem import io
+from expertsystem.topology import (
+    Edge,
+    InteractionNode,
+    SimpleStateTransitionTopologyBuilder,
+    Topology,
+)
+
+
+@pytest.fixture(scope="package")
+def four_body_decay() -> Topology:
+    r"""Create a dummy `Topology`.
+
+    Has the following shape:
+
+    .. code-block::
+
+        e0 -- (N0) -- e1 -- (N1) -- e3 -- (N2) -- e5
+                \             \             \
+                 e2            e4            e6
+    """
+    topology = Topology(
+        nodes={0, 1, 2},
+        edges={
+            0: Edge(0, None),
+            1: Edge(1, 0),
+            2: Edge(None, 0),
+            3: Edge(2, 1),
+            4: Edge(None, 1),
+            5: Edge(None, 2),
+            6: Edge(None, 2),
+        },
+    )
+    io.dot.write(topology, "four_body_decay.gv")
```

Comment:
why the write?

Reply:
Just a helper in case you want to visualise

Reply:
Hmm i would separate that though (like one fixture depending on another), is it actually needed right now?

Reply:
Nah it was just a trick for me to check whether it looked like what I had in mind :grimacing:

Reply:
Then i would just remove it

Reply:
8b83b09a